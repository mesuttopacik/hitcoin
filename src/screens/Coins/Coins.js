import {View, ActivityIndicator, FlatList} from 'react-native';
import React, {useState, useEffect} from 'react';
import Error from '../../components/Error';
import CoinCard from '../../components/CoinCard';

import { useDispatch, useSelector } from 'react-redux'
import { fetchCoins } from '../../redux/listSlice'
import { coinSelected } from '../../redux/detailsSlice'

const Coins = ({navigation}) => {
  const dispatch = useDispatch();
  const { coins, status, error } = useSelector((state) => state.list)
  
  useEffect(() => {
    dispatch(fetchCoins(coins.length));
  }, []);

  const coinClickHandler = coin => {
    dispatch(coinSelected(coin));
    navigation.navigate('Detail', {coin});
  };

  const renderCoins = ({item}) => {
    return (
      <CoinCard item={item} onClick={() => coinClickHandler(item)} />
    )
  };

  if (error) {
    return <Error err={error} />;
  }

  if (status == "loading") {
    return <ActivityIndicator size="large" />;
  }

  const loadMore = () => {
    dispatch(fetchCoins(coins.length));
  };

  return (
    <View>
      <FlatList
        data={coins}
        renderItem={renderCoins}
        onEndReachedThreshold={0.2}
        keyExtractor={item => item.id}
        onEndReached={loadMore}
      />
    </View>
  );
};
export default Coins;
