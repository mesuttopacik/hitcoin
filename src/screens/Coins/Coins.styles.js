import { StyleSheet } from 'react-native'
import { colors } from '../../constants/GlobalStyles'

export default StyleSheet.create({
    container:{
        backgroundColor:colors.ContentBackground
    }
})