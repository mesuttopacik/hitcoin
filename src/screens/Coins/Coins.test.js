import React from 'react';
import {render} from '@testing-library/react-native';
import Coins from './Coins'


test('should add coins to list', () => { 
    const comp = render(<Coins />);
    expect(comp).toMatchSnapshot();
 })

