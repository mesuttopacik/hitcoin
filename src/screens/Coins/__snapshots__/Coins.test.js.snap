// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add coins to list 1`] = `
<ActivityIndicator
  size="large"
/>
`;
